cmd/snap-confine/mount-support.c

```diff
@@ -404,6 +404,64 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 			    dst);
 		}
 	}
+	// Since we mounted /etc from the host filesystem the /etc/os-release file
+	// (which may not be present) may be a symbolic link. It is expected that
+	// we see the real /etc/os-release file so let's do our best to show
+	// /etc/os-release as it looks like on the classic distribution.
+	//
+	// If the symbolic link in /etc is pointing to /usr/lib/os-release then
+	// bind mount /usr/lib/os-release over itself in the core snap. This way
+	// the symlink (which we cannot be the target of a bind mount as the kernel
+	// will always follow the link when performing the mount operation) will
+	// work as expected.
+	//
+	// https://bugs.launchpad.net/canonical-livepatch-client/+bug/1667470
```

Comment:
In thinking about this more, what should /etc/os-release actually be? Is it something for snaps to figure out what 'type: os' snaps should be? Ie, should /etc/os-release be different for a 'type: app' snap built for the Ubuntu 'core' snap and a 'type: app' snap built for Fedora's 'core' snap? If so, then the current implementation is correct and this PR is wrong. If not, how will a 'type: app' snap built for Fedora by expected to work if it was built with Fedora in mind and all of a sudden os-release says Ubuntu because someone is running a Fedora snap and core runtime on classic Ubuntu? On a classic 14.04 Ubuntu system, how meaningful is it for snaps built for 16.04 to see 14.04 in the os-release?

Reading https://www.freedesktop.org/software/systemd/man/os-release.html it feels to me like having os-release reflect the core snap's runtime (ie, the current implementation) is the most correct, because from the perspective of the snap, the runtime is the operating system. If we agree on that, then we need only decide on how livepatch should determine how it should run or not, which can be discussed elsewhere (though it seems that what it is most interested in is if a system is live patchable, therefore perhaps simply examining /proc/version_signature is enough. If this is not enough, it seems reasonable for the snapd rest api to be able to tell snaps the host's os-release; or alternatively, figure out a way to take advantage of the precedence of /etc/os-release over /usr/lib/os-release that is defined in https://www.freedesktop.org/software/systemd/man/os-release.html; or create a `os-release.system` (or similar) file somewhere that livepatch can examine).